### PR TITLE
fix: infinity loop and improve farm performance

### PR DESCRIPTION
**Pull request by namgold**

### CODE REVIEW

File: src/state/application/hooks.ts [hunk 1]
```diff
@@ -320,23 +320,29 @@ export function useETHPrice(version: string = VERSION.CLASSIC): AppState['applic
...
-              oneDayBackPrice: (oneDayBackPrice ? oneDayBackPrice : 0).toString(),
-              pricePercentChange,
-            })
-          : updateETHPrice({
-              currentPrice: (newPrice ? newPrice : 0).toString(),
-              oneDayBackPrice: (oneDayBackPrice ? oneDayBackPrice : 0).toString(),
-              pricePercentChange,
-            }),
-      )
+      if (fetchingETHPrice) return
```
Comment (viet-nv) [inline src/state/application/hooks.ts:323]: why dont you use a loading state instead of this?

Comment (viet-nv) [reply src/state/application/hooks.ts:323]: oh, multiple components use this hook simultaneously

Comment (namgold) [reply src/state/application/hooks.ts:323]: Yes. Multiple component call this and it just need to load once.

File: src/state/application/hooks.ts [hunk 2]
```diff
@@ -433,7 +439,7 @@ export function useTokensPrice(tokens: (Token | NativeCurrency | null | undefine
   const [prices, setPrices] = useState<number[]>([])
   const { elasticClient, classicClient } = useKyberSwapConfig()
 
-  useDeepCompareEffect(() => {
```
Comment (namgold) [inline src/state/application/hooks.ts:436]: this make infinity loop

File: src/state/farms/classic/updater.ts [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+      checkForFarms()
+    }, 30_000)
+
+    return () => {
+      cancelled = true
+      clearInterval(i)
+    }
+  }, [dispatch, chainId, fairLaunchContracts, account, isEVM, networkInfo, classicClient, blockClient, allTokens])
+
+  useDebug({
```
Comment (nguyenhoaidanh) [inline src/state/farms/classic/updater.ts:232]: rm

**Review by viet-nv — APPROVED** [review]

File: src/components/YieldPools/index.tsx [hunk 4]
```diff
@@ -48,12 +53,12 @@ const YieldPools = ({ loading, active }: { loading: boolean; active?: boolean })
       if (farm.rewardPerSeconds) {
         // for active/ended farms
         return (
-          currentTimestamp &&
+          currentTimestampRef.current &&
```
[the remark this answers is not in the document]
Comment (namgold) [reply src/components/YieldPools/index.tsx:56]: It not only use once. It's been used in 2 function.

**Pull request merged**